discord/ext/commands/view.py

```diff
@@ -108,21 +108,50 @@ def __repr__(self):
 
 # Parser
 
+# map from opening quotes to closing quotes
+quotes = {
+    '"': '"',
+    "'": "'",
+    "«": "»",
+    "‘": "’",
+    "‚": "‛",
+    "“": "”",
+    "„": "‟",
+    "‹": "›",
+    "⹂": "⹂",
+    "「": "」",
+    "『": "』",
+    "〝": "〞",
+    "﹁": "﹂",
+    "﹃": "﹄",
+    "＂": "＂",
+    "＇": "＇",
+    "｢": "｣",
+    "«": "»",
+    "‹": "›",
+    "《": "》",
+    "〈": "〉",
+}
+all_quotes = set(quotes.keys()) | set(quotes.values())
+
 def quoted_word(view):
     current = view.current
 
     if current is None:
         return None
 
-    is_quoted = current == '"'
+    is_quoted = current in quotes
+    if is_quoted:
+        open_quote = current
+        close_quote = quotes[current]
```

Comment:
I don't like this implementation at all.

You're essentially doing two lookups in the dictionary (once for `in` and a second for `quotes[current]`). Instead you should opt for `quotes.get(current)` which will do both at a time (it defaults to `None` is none is found which is false-y).

Reply:
I'll implement that

Reply:
So, I changed the implementation, but I decided to run a quick test and it looks like using `in` and a lookup is actually faster than using `.get()`. Here's the script I used to test:

```
from timeit import timeit

setup = '''
quotes = {'"': '"', "'": "'", "«": "»", "‘": "’", "‚": "‛", "“": "”", "„": "‟", "‹": "›", "⹂": "⹂", "「": "」", "『": "』", "〝": "〞", "﹁": "﹂", "﹃": "﹄", "＂": "＂", "＇": "＇", "｢": "｣", "«": "»", "‹": "›", "《": "》", "〈": "〉"}

def get():
    a = quotes.get('"')
    b = quotes.get('a')

def in_():
    a = quotes['"'] if '"' in quotes else None
    b = quotes['a'] if 'a' in quotes else None
'''

print('Using get():', timeit('get()', setup=setup))
print('Using in {}:', timeit('in_()', setup=setup))
```